plio/views.py

```diff
@@ -183,17 +177,38 @@ def list_uuid(self, request):
 
         num_plios = queryset.count()
         queryset = self.filter_queryset(queryset)
-
-        uuid_list = queryset.values_list("uuid", flat=True)
-        page = self.paginate_queryset(uuid_list)
+        page = self.paginate_queryset(queryset.values())
 
         if page is not None:
+            # find the number of unique views
+            plio_ids = list(queryset.values_list(flat=True))
+
+            unique_user_counts = (
+                Session.objects.filter(plio__in=plio_ids)
+                .values("plio")
+                .annotate(count=Count("user", distinct=True))
+            ).order_by()
+
+            plio_id_to_count = {
+                unique_user_count["plio"]: unique_user_count["count"]
+                for unique_user_count in unique_user_counts
+            }
+
+            # add the number of unique viewers and items corresponding
+            # to the plio in each plio object
+            for index, plio in enumerate(page):
+                page[index]["num_views"] = plio_id_to_count.get(plio["id"], 0)
+                page[index]["items"] = ItemSerializer(
+                    queryset[index].item_set, many=True
+                ).data
+
```

Comment:
A couple of things here
- You're calculating unique viewers for each plio here and saving that in a temporary dictionary and attaching that information in the response, right? In a file called `ordering.py`, there is already a piece of code jisme I calculated unique viewers for each plio and save that information in the queryset itself. I think it would be best to reuse code in some way, as it does the same thing. 
- Also, when I do just `list`, this piece of code runs and I get number of unique views for each of the 5 plios. But let's suppose after that I want to order by views, so I click the views column in the table in frontend. That makes a call to `list` again but this time with the ordering filter. So what happens is that the compiler goes to `ordering.py`, calculates the unique views for each plio, sorts them according to that, and then AGAIN finds the num unique viewers for each plio (the code that you wrote). So double work is being done here. After the compiler gets through `ordering.py`, the information of num unique viewers is already stored in the queryset so you can reuse that.